service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java

```diff
@@ -0,0 +1,119 @@
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.JUDGE_AMENDS_DRAFT;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.SEND_TO_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ReviewCMOService {
+
+    private final ObjectMapper mapper;
+    private final Time time;
+
+    private DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos) {
+        return buildDynamicList(cmos, null);
+    }
+
+    public DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos, UUID selected) {
+        return asDynamicList(cmos, selected, uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder::getHearing);
+    }
+
+    public Map<String, Object> handlePageDisplayLogic(CaseData caseData) {
+        List<Element<CaseManagementOrder>> cmosReadyForApproval = getCMOsReadyForApproval(
+            caseData.getDraftUploadedCMOs());
+        Map<String, Object> data = new HashMap<>();
+        String numDraftCMOs = "numDraftCMOs";
+        
+        switch (cmosReadyForApproval.size()) {
+            case 0:
+                data.put(numDraftCMOs, "NONE");
+                break;
+            case 1:
+                CaseManagementOrder cmo = cmosReadyForApproval.get(0).getValue();
+                data.put(numDraftCMOs, "SINGLE");
+                data.put("reviewCMODecision",
+                    ReviewDecision.builder().hearing(cmo.getHearing()).document(cmo.getOrder()).build());
+                break;
+            default:
+                data.put(numDraftCMOs, "MULTI");
+                data.put("cmoToReviewList", buildDynamicList(cmosReadyForApproval));
+                break;
+        }
+
+        return data;
+    }
+
+    public Element<CaseManagementOrder> getCMOToSeal(CaseData caseData) {
+        Element<CaseManagementOrder> cmo = getSelectedCMO(caseData);
+        DocumentReference order;
+
+        if (JUDGE_AMENDS_DRAFT.equals(caseData.getReviewCMODecision().getDecision())) {
+            order = caseData.getReviewCMODecision().getJudgeAmendedDocument();
+        } else {
+            order = cmo.getValue().getOrder();
+        }
+        return element(cmo.getValue().toBuilder()
+            .dateIssued(time.now().toLocalDate())
+            .status(CMOStatus.APPROVED)
+            .order(order)
+            .build());
+    }
+
+    public List<Element<CaseManagementOrder>> getCMOsReadyForApproval(List<Element<CaseManagementOrder>> draftCMOs) {
```

Comment:
can it be changed to accept CaseData as param, now you dont control if client sent correct thing,

all execution of this method looks like:
reviewCMOService.getCMOsReadyForApproval(
            caseData.getDraftUploadedCMOs())

but can be 
reviewCMOService.getCMOsReadyForApproval(caseData)

It will make interface of this service more consistent as most of rest methods accept CaseData